### PR TITLE
Wr/parking orbit3

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
parking-orbit `plugin-templates` and `templates` library 🚀 
ran `sf-install` in each package
added a few types
resolved core3 updates

### What issues does this PR fix or reference?
@W-11445939@

### CODE REVIEW

File: packages/plugin-templates/src/utils/messageUtil.ts [hunk 1]
```diff
@@ -35,10 +35,7 @@ export class MessageUtil {
     extra?: string
   ): string {
     return (
-      this.messages.getMessage(descriptionKey, tokens) +
-      '\n' +
-      this.getHelpHead(isLightningBundle) +
-      (extra || '')
+      this.messages.getMessage(descriptionKey, tokens) + '\n' + this.getHelpHead(isLightningBundle) + (extra || '')
```
Comment (mshanemc) [inline packages/plugin-templates/src/utils/messageUtil.ts:38]: nice approach, putting messages behind the existing util class

File: packages/templates/package.json [hunk 2]
```diff
@@ -12,36 +12,49 @@
...
-    "ts-node": "^8.6.2",
+    "husky": "^7.0.4",
+    "mocha": "^9.1.3",
+    "nyc": "^15.1.0",
+    "prettier": "^2.0.5",
+    "pretty-quick": "^3.1.0",
+    "shx": "^0.3.4",
+    "sinon": "10.0.0",
+    "ts-node": "^10.0.0",
     "typescript": "^4.3.2",
```
Comment (mshanemc) [inline packages/templates/package.json:58]: it's a bit odd to have different versions of TS in the different packages?

File: packages/templates/package.json [hunk 3]
```diff
@@ -12,36 +12,49 @@
...
   "devDependencies": {
-    "@types/chai": "^4",
+    "@salesforce/dev-scripts": "^2.0.4",
+    "@salesforce/dev-config": "^3.0.0",
+    "@salesforce/prettier-config": "^0.0.2",
     "@types/chai-as-promised": "^7.1.4",
     "@types/fs-extra": "^8.1.0",
     "@types/got": "^9.6.12",
     "@types/mime-types": "^2.1.0",
     "@types/mkdirp": "0.5.2",
```
Comment (mshanemc) [inline packages/templates/package.json:31]: what needs this?

File: packages/templates/src/service/gitRepoUtils.ts [hunk 4]
```diff
@@ -35,13 +35,9 @@ export async function getRepoInfo(repoUri: URL): Promise<RepoInfo> {
 
   // For repos with no branch information, fetch default branch
   if (t === undefined) {
-    const infoResponse = await got(
```
Comment (mshanemc) [inline packages/templates/src/service/gitRepoUtils.ts:38]: this is not gonna be proxy-friendly!  see https://github.com/salesforcecli/plugin-info/pull/75/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

Anytime we're connecting to something outside of a salesforce org (which should go through jsforce connection) we need to check for proxies.  

[I'm gonna figure out how to make this a linter rule eventually, too]

**Review by mshanemc — CHANGES_REQUESTED** [review]
let's fix the proxy thing while we're here

File: packages/plugin-templates/src/utils/templateCommand.ts [hunk 5]
```diff
@@ -49,9 +45,7 @@ export abstract class TemplateCommand extends SfdxCommand {
   public static async getApiVersion(): Promise<string> {
     try {
       const aggregator = await TemplateCommand.getConfigAggregator();
-      const apiVersionFromConfig = aggregator.getPropertyValue(
-        TemplateCommand.API_VERSION
-      ) as string;
+      const apiVersionFromConfig = aggregator.getPropertyValue(TemplateCommand.API_VERSION) as string;
```
Comment (randi274) [inline packages/plugin-templates/src/utils/templateCommand.ts:48]: should we use `OrgConfigProperties.ORG_API_VERSION` for consistency here?

**Review by randi274 — COMMENTED** [review]
Phew PR of all formatting changes! Some easy nits to fix, and a few questions. Still need to plug in and play around 🔌

File: packages/plugin-templates/command-snapshot.json [hunk 6]
```diff
@@ -1,174 +1,169 @@
...
-            "loglevel",
-            "outputdir",
-            "plugins",
-            "hidden",
-            "erroronwarnings"
-        ]
-    },
     {
         "command": "force:analytics:template:create",
+        "plugin": "@salesforce/plugin-templates",
```
[the remark this answers is not in the document]
Comment (WillieRuemmele) [reply packages/plugin-templates/command-snapshot.json:4]: there've been a few changes to the snapshot in the new major version, we now track which plugin a command belongs to, and any aliases they have. This allows us to ensure no breaking changes at the cli level

**Comment by randi274** [comment]
@smithgp I added you as a reviewer on this as a heads-up for the Analytics plugin. Not sure if this might impact y'all, but the big changes are updating the command library, and also the [core v2-v3](https://github.com/forcedotcom/sfdx-core/blob/v3/MIGRATING_V2-V3.md) migration.

**Review by randi274 — APPROVED** [review]
Woo! Great work getting this moved over. QA notes are linked in the Quip tab added to the WI. 

You're the Templates Ninja now @WillieRuemmele 🥷 🌟

**Review by mshanemc — APPROVED** [review]

**Pull request merged**